### ISSUE TITLE
virtualbuildenv generator does not set LIB env var correctly in activate_build.bat

### ISSUE THREAD

**Issue by jmoutray** [conan-io/conan#4582]
Running **conan install** with -g virtualbuildenv generates activate_build.bat.
The first four env vars set are:
```
SET PROMPT=...
SET CL=...
SET LIB=...
SET_LINK_=...
```
CL and \_LINK\_ are set correctly, but LIB does not contain my "lib_path".

According to the following code, what happens is LIB added to a dictionary with the correct lib_path, but then the call to "update" to add "vc_vars" overwrites the LIB value with the Visual Studio LIB value.

**virtualbuildenv.py**
```javascript
        if compiler == "Visual Studio":
            self.env = VisualStudioBuildEnvironment(conanfile).vars_dict
            self.env.update(vcvars_dict(conanfile.settings, output=conanfile.output))
        else:
            self.env = AutoToolsBuildEnvironment(conanfile).vars_dict
```
This is incorrect.  The lib_path needs to be merged with the Visual Studio LIB value instead of being replaced.

I'm using conan v1.12.3 on Windows 10 and Visual Studio 2015.

**Comment by memsharded** [comment]
Yes, I'd say you are right, this seems a bug.

I think the solution could be something like https://github.com/conan-io/conan/pull/4583 (missing some test to validate it):

```python
        if compiler == "Visual Studio":
            self.env = VisualStudioBuildEnvironment(conanfile).vars_dict
            settings_vars = vcvars_dict(conanfile.settings, output=conanfile.output)
            for env_var in self.env:
                self.env[env_var].extend(settings_vars.pop(env_var, []))
            self.env.update(settings_vars)
        else:
            self.env = AutoToolsBuildEnvironment(conanfile).vars_dict
````

**Comment by memsharded** [comment]
This has been fixed and will be released in conan 1.13, keep tuned :)
